test/python/sql_validation.py

```diff
@@ -0,0 +1,453 @@
+        with open(i, "r") as testfile:
+            testline = testfile.readline()
+
+            while testline:
+                testline = testline.strip()
+
+                # ignoring comments in the file
+                if testline.startswith("--") or testline.startswith("/*") or testline.startswith("#"):
+                    testline = testfile.readline()
+                    continue
+
+                # if pattern exists, return true
+                elif(re.search( obj_name, testline, re.I)):
+                    return True
+                testline = testfile.readline()
+    return False
+
+
+# find the object name in the test files
+def find_in_testfiles(type, searchinfiles, object_names, f_path, logger):
+
+    with open(f_path, "a") as expected_file:
+        for object in object_names:
+
+            # flag for object name found or not in the JDBC input files
+            flag = False
+            object_name = object[1]
+
+            # replacing . as in regex it means any character
+            if "." in object[1]:
+                object_name = object[1].replace(".", "[.]")
+
+            # adding word boundary if object is not a function
+            if not object[0] == 'function':
+                object_name = object_name + r"\b"
+
+            flag = find_obj(searchinfiles, r"\b" + object_name)
+
+            # searching again without schema name for objects in all schemas
+            if(flag == False and "." in object[1]):
+                object_name = r"\b" + object_name.split(".]")[-1]
+                
+                # creating pattern to identify @@func calls
+                if object[0] == 'function':
+                    object_name = "(" + "@@" + object_name.replace("\s{0,2}[(]", "").replace("\\b", "") + ")|(" +  object_name + ")"
+                
+                flag = find_obj(searchinfiles, object_name)
+            
+            # if tests not found, add it to output file
+            if flag == False:
+                expected_file.write("Could not find {0}tests for {1} {2}\n".format(type, object[0], object[1].split('\s')[0]))
+    logger.info("{}Tests for objects found successfully!".format(type))
+
+
+# creates list of all files to be searched for upgrade test
+def list_upgrade_files():
+    upgrade_files = []
+
+    inpPath = "../JDBC/upgrade"
+
+    # searching prepare and verify scripts in upgrade directory
+    update_files = list_files(inpPath, "**/*[-][v][u][-][!c]*.*")
+    upgrade_files.extend(update_files)
+
+    # list all schedule files
+    sch_files = list_files(inpPath, "**/schedule")
+
+    # create a set of test files to be looked into JDBC input
+    search_tests = set()
+    for filename in sch_files:
+        with  open(filename, "r") as file:
+            line = file.readline()
+
+            while line:
+                line = line.strip()
+
+                # ignoring comments and empty lines in the schedule file
+                if line.startswith("#") or line == '':
```

Comment:
We need to skip some more patterns as well which we are skipping here https://github.com/babelfish-for-postgresql/babelfish_extensions/blob/BABEL_2_X_DEV/.github/composite-actions/setup-base-version/action.yml#L81